gap/AlmostSimpleDefiningCharacteristic.gi

```diff
@@ -0,0 +1,26 @@
+# Construction of group as in Table 5.6 row 4 from [BHR13]
+BindGlobal("l3qdim6",
+function(q)
+    local A, G, M, MM, S, T, general, w;
+    general := ValueOption("general");
```

Comment:
The `ValueOption` abomination should be replaced by adding a variadic argument. This whole option business accesses and modifies a global state, from the docs `8 Options Stack`: "GAP  supports a global options system".

Reply:
Why does it even have to be variadic? Can't this function simply have two arguments, the second being `general`,l and people are free to pass in `false` for it?

BTW, while I agree options are problematic, they are often used to substitute for a feature GAP sadly is lacking: keyword arguments. In GAP you can end up with function calls like `func(true, true, false)` that can be rather confusing, compared to something like `func(general:=true, checks:=true, format_drive:=false)`

Reply:
I guess what I tried to say with my last remark is: dang, we really should have keyword arguments in GAP ;-). I once started working on that but quickly decided to put it aside, because while I think implementing it would be fun, I really should be spending my time on other things *sigh*. But if anybody is ever interested in diving into the GAP kernel and implementing some language extensions, I'd be happy to teach you the ropes.